### PR TITLE
[ONNX] Add pixel_unshuffle support in opset 9

### PR DESCRIPTION
**Pull request by shubhambhokare1**
Current we are unable to utilize ONNX's SpaceToDepth operator due to the lack of the mode_s attribute, hence we add an alternative symbolic in opset 9 to support pixel_unshuffle

- Adds support for pixel_unshuffle in opset9
- Adds support for dynamic input shapes for pixel_shuffle and pixel_unshuffle

### CODE REVIEW

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -2159,6 +2159,29 @@ def pixel_shuffle(g, self, upscale_factor):
                                     allowzero=0)
 
 
+@parse_args("v", "i")
+def pixel_unshuffle(g, self, downscale_factor):
+    dims = sym_help._get_tensor_sizes(self)
+    if len(dims) != 4:
+        return _unimplemented("pixel_shuffle", "only support 4d input")
+    if any([i is None for i in dims[1:]]):
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2167]: nit
```suggestion
    if any(i is None for i in dims[1:]):
```

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -2159,6 +2159,29 @@ def pixel_shuffle(g, self, upscale_factor):
...
+        return _unimplemented("pixel_shuffle", "only support static input shape, except for batch size")
+    output_channel = dims[1] * downscale_factor * downscale_factor
+    after_view = sym_help._reshape_helper(g, self,
+                                          g.op("Constant", value_t=torch.tensor([-1, dims[1],
+                                                                                dims[2] // downscale_factor,
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2172]: are dims[2] and dims[3] guaranteed to be divisible by downscale_factor?

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -2159,6 +2159,29 @@ def pixel_shuffle(g, self, upscale_factor):
...
+@parse_args("v", "i")
+def pixel_unshuffle(g, self, downscale_factor):
+    dims = sym_help._get_tensor_sizes(self)
+    if len(dims) != 4:
+        return _unimplemented("pixel_shuffle", "only support 4d input")
+    if any([i is None for i in dims[1:]]):
+        return _unimplemented("pixel_shuffle", "only support static input shape, except for batch size")
+    output_channel = dims[1] * downscale_factor * downscale_factor
+    after_view = sym_help._reshape_helper(g, self,
+                                          g.op("Constant", value_t=torch.tensor([-1, dims[1],
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2171]: would 0 be better than -1 for retaining batch size?

File: torch/onnx/symbolic_opset9.py [hunk 4]
```diff
@@ -2159,6 +2159,29 @@ def pixel_shuffle(g, self, upscale_factor):
                                     allowzero=0)
 
 
+@parse_args("v", "i")
+def pixel_unshuffle(g, self, downscale_factor):
+    dims = sym_help._get_tensor_sizes(self)
+    if len(dims) != 4:
+        return _unimplemented("pixel_shuffle", "only support 4d input")
+    if any([i is None for i in dims[1:]]):
+        return _unimplemented("pixel_shuffle", "only support static input shape, except for batch size")
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:2168]: had an idea that possible can work for all dynamic input shape. maybe reshape 1 dim at a time with 0s to divide the height/weight, like `first_view = reshape(input, [0, 0, -1, scale, 0])`, then `second_view = reshape(input, [0, 0, 0, 0, -1, scale])`.

**Review by BowenBao — COMMENTED** [review]
Looks good overall, some minor comments, and suggestion for possible dynamic case support (if necessary).

File: torch/onnx/symbolic_opset9.py (hunk 2 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:2172]: Yup the shape of the input is 
(*, C, dims[2] * downscale_factor,  dims[3] * downscale_factor)
So should be always divisible

File: torch/onnx/symbolic_opset9.py (hunk 4 above)
Comment (shubhambhokare1) [reply torch/onnx/symbolic_opset9.py:2177]: Is there any way we could force the division to result in being an integer?
From what I understand, seems like [0, 0, -1, scale , 0] is essentially [0, 0, dim/scale, scale, 0] however we want the division to be 
```dim // scale```, otherwise we would get a floating point exception

**Review by BowenBao — APPROVED** [review]
